### PR TITLE
[k8s] Add missed env variables for EA

### PR DESCRIPTION
**Pull request by dmolokanov**

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs [hunk 1]
```diff
@@ -248,14 +247,6 @@ private async Task UpsertDeployments(V1ServiceList currentServices, V1Deployment
                     // Get the converted pod
                     V1PodTemplateSpec v1PodSpec = kubernetesModelBuilder.GetPod();
 
-                    // if this is the edge agent's deployment then it needs to run under a specific service account
```
Comment (kkdawkins) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs]: Don't we need this if we are launching EA?

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs [hunk 2]
```diff
@@ -219,6 +216,8 @@ private async Task UpsertDeployments(V1ServiceList currentServices, V1Deployment
             foreach (KubernetesModule<TConfig> module in customObject.Spec)
             {
                 var moduleId = moduleIdentities[module.Name];
+
+                string deploymentName = this.DeploymentName(moduleIdentities[module.Name].ModuleId);
```
Comment (kkdawkins) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs]: Where is deploymentName being used?

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesPodBuilder.cs [hunk 3]
```diff
@@ -68,10 +68,12 @@ public V1PodTemplateSpec GetPodFromModule(Dictionary<string, string> labels, IMo
             // Image
             string moduleImage = module.Config.Image;
 
+            var name = KubeUtils.SanitizeDNSValue(moduleIdentity.ModuleId);
```
Comment (kkdawkins) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesPodBuilder.cs]: Why is this necessary? KubeUtils is a simple extension method - no need for it's own var.

Comment (dmolokanov) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesPodBuilder.cs]: why sanitize data multiple time instead of one time and reuse multiple times?

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs (hunk 1 above)
Comment (dmolokanov) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs]: nope. iotedged provide the value for EA

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs (hunk 2 above)
Comment (dmolokanov) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs]: yeah, should be used in labels

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesPodBuilder.cs (hunk 3 above)
Comment (kkdawkins) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesPodBuilder.cs]: I see. You added "name" to V1ObjectMeta. At leat name it better...

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs (hunk 1 above)
Comment (kkdawkins) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs]: Wouldn't that have been through SA name? I am confused why this was needed before and no longer needed - can you elaborate on what changed?

Comment (dmolokanov) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs]: It was needed before because only EA had a service account with a special name. Now every module will have its own service account that equals sanitized module name. So no need to pass env variable for it anymore

Comment (kkdawkins) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs]: That makes sense!

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs [hunk 4]
```diff
@@ -223,7 +223,7 @@ private async Task UpsertDeployments(V1ServiceList currentServices, V1Deployment
                     // Default labels
                     var labels = new Dictionary<string, string>
                     {
-                        [Constants.K8sEdgeModuleLabel] = KubeUtils.SanitizeLabelValue(moduleId.ModuleId),
+                        [Constants.K8sEdgeModuleLabel] = deploymentName,
```
Comment (darobs) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs]: What's the difference between module id and deployment name?

Comment (dmolokanov) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs]: moduleId is module identity. 
deployment name is module name sanitized to k8s name rules.
